.github/workflows/ci.yml

```
@@ -13,72 +13,49 @@ on:
+    runs-on: ubuntu-latest
+    outputs:
+      pmap: ${{ steps.crds-context.outputs.pmap }}
     steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python-version }}
-          cache: 'pip'
-          cache-dependency-path: 'pyproject.toml'
-      - run: pip install "tox>=4.0"
-      - run: tox -e ${{ matrix.toxenv }}
+      - run: >
+          echo "pmap=$(
+          curl -s -X POST -d '{"jsonrpc": "1.0", "method": "get_default_context", "params": ["jwst"], "id": 1}' ${{ env.CRDS_SERVER_URL }}/json/ |
+          python -c "import sys, json; print(json.load(sys.stdin)['result'])"
+          )" >> $GITHUB_OUTPUT
+        # Get default CRDS_CONTEXT without installing crds client
+        # See https://hst-crds.stsci.edu/static/users_guide/web_services.html#generic-request
+        id: crds-context
   test:
-    name: ${{ matrix.toxenv }} (Python ${{ matrix.python-version }}, ${{ matrix.os }})
-    runs-on: ${{ matrix.os }}
-    env:
-      CRDS_PATH: /home/runner/work/stdatamodels/crds_cache
-      CRDS_CLIENT_RETRY_COUNT: 3
-      CRDS_CLIENT_RETRY_DELAY_SECONDS: 20
-    strategy:
-      fail-fast: false
-      matrix:
-        toxenv: [ test-xdist ]
-        python-version: [ '3.8', '3.9', '3.10', '3.11' ]
-        os: [ ubuntu-latest, macos-latest ]
-        include:
-          - toxenv: test-xdist-cov
-            os: ubuntu-latest
-            python-version: '3.11'
-          - toxenv: test-jwst-xdist-cov
-            os: ubuntu-latest
-            python-version: '3.11'
-    steps:
-      - if: ${{ contains(matrix.toxenv,'docs') }}
-        run: sudo apt-get install graphviz texlive-latex-extra dvipng
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python-version }}
-          cache: 'pip'
-          cache-dependency-path: 'pyproject.toml'
-      - run: echo "CRDS_PATH=$HOME/crds_cache" >> $GITHUB_ENV
-      - run: echo "CRDS_SERVER_URL=https://jwst-crds.stsci.edu" >> $GITHUB_ENV
-      - run: pip install crds
-      - run: echo "pmap=$(crds list --operational-context)" >> $GITHUB_OUTPUT
-      - uses: actions/cache@v3
-        with:
-          path: ${{ env.CRDS_PATH }}
-          key: crds-${{ steps.crds-context.outputs.pmap }}
-      - run: crds sync --contexts ${{ steps.crds-context.outputs.pmap }}
-      - run: pip install "tox>=4.0"
-      - run: tox -e ${{ matrix.toxenv }}
-      - if: contains(matrix.toxenv,'-cov')
-        uses: codecov/codecov-action@v3
-        with:
-          file: ./coverage.xml
-          flags: unit
-          fail_ci_if_error: true
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
+    needs: [ crds_context ]
+    with:
+      cache-path: /tmp/crds_cache
+      cache-key: crds-${{ needs.crds_context.outputs.pmap }}
```

I've opened a PR (https://github.com/OpenAstronomy/github-actions-workflows/pull/107) to add support for env vars being passed through. I want to get more feedback on the new API before merging, but you can try it out on my branch if you like:
```suggestion
    uses: ConorMacBride/github-actions-workflows/.github/workflows/tox.yml@set-env-tox
    needs: [ crds_context ]
    with:
      cache-path: /tmp/crds_cache
      cache-key: crds-${{ needs.crds_context.outputs.pmap }}
      setenv: |
        CRDS_PATH: /tmp/crds_cache
```